internal/querycoordv2/balance/balance.go

```diff
@@ -119,22 +120,29 @@ func (b *RoundRobinBalancer) AssignChannel(ctx context.Context, collectionID int
 	if len(nodesInfo) == 0 {
 		return nil
 	}
+
+	plans := make([]ChannelAssignPlan, 0)
+	scoreDelta := make(map[int64]int)
+	if streamingutil.IsStreamingServiceEnabled() {
+		channels, plans, scoreDelta = assignChannelToWALLocatedFirstForNodeInfo(channels, nodesInfo)
+	}
```

Comment:
For one channel,
only one replica of collection will always have a wal located sqn, so the following channel policy doesn't work.
Other replica can not hit the wal located sqn, so the only channel policy works.

Make a boost priority here is hard to achieve previous logic, because different channel has different boost for different sqn.